pkg/k8s/watchers/endpoints.go

```diff
@@ -138,31 +143,22 @@ func (k *K8sEndpointsWatcher) addKubeAPIServerServiceEndpoints(eps *k8s.Endpoint
 //
 // The actual implementation of this logic down to the datapath is handled
 // asynchronously.
-func (k *K8sEndpointsWatcher) handleKubeAPIServerServiceEPChanges(desiredIPs map[cmtypes.PrefixCluster]struct{}, rid ipcacheTypes.ResourceID) {
+func (k *K8sEndpointsWatcher) handleKubeAPIServerServiceEPChanges(desiredIPs set.Set[cmtypes.PrefixCluster], rid ipcacheTypes.ResourceID) {
 	src := source.KubeAPIServer
 
-	// We must perform a diff on the ipcache.identityMetadata map in order to
-	// figure out which IPs are stale and should be removed, before we inject
-	// new IPs into the ipcache. The reason is because kube-apiserver will
-	// constantly reconcile this specific object, even when it's been deleted;
-	// effectively, this means we can avoid listening for the delete event.
-	// Therefore, any changes to this specific object can be handled in a
-	// "flattened" manner, since the most up-to-date form of it will be an add
-	// or update event. The former is sent when Cilium is syncing with K8s and
-	// the latter is sent anytime after.
-	//
-	// For example:
-	//   * if a backend is removed or updated, then this will be in the form of
-	//     an update event.
-	//   * if the entire object is deleted, then it will quickly be recreated
-	//     and this will be in the form of an add event.
-	k.ipcache.RemoveLabelsExcluded(
-		labels.LabelKubeAPIServer,
-		desiredIPs,
-		rid,
-	)
+	// remove stale IPs
+	for ip := range k.existingAPIServers.Members() {
```

Comment:
Is this really equivalent to what was there before?
Here, we remove the label from all existing and not desired IPs and below add it to all desired but not known IPs. But the previous logic would look through _all_ IPs and remove the label from there unless desired. Can there not be some way (maybe restoration?) where an IP has that label without being present in either existing nor desired?